docs/privacyidea.md

```diff
@@ -216,6 +216,48 @@ Use the following example:
         'excludeClientIPs'  => array("10.0.0.0-10.2.0.0", "192.168.178.2"),
     ),
 
+    /**
+     *  This filter is optional. If you want to selectively disable the privacyIDEA authentication using the
+     *  the entityID and/or SAML attributes, you may enable this filter
+     */
+    22 => array(
+        'class'                => 'privacyidea:checkEntityID',
+        /**
+         *  Depending on entityids and excludeattributes the filter will set the state variable 
+         *  $state[$setPath][$setPath] to true or false.
+         *  To selectively enable or disable privacyIDEA, make sure that you specify setPath and setKey such
+         *  that they equal enabledPath and enabledKey from privacyidea:serverconfig or privacyidea:privacyidea.
+         */
+        'setPath'              => 'privacyIDEA',
+        'setKey'               => 'enabled',
+        /**
+         *  The requesting SAML provider's entityID will be tested against this list of regular expressions.
+         *  If there is a match, the filter will set the specified state variable to false. The first matching 
+         *  expression will take precedence.
+         */
+        'entityids'            => array(
+                                      '/http(s)\/\/conditional-no2fa-provider.de\/(.*)/',
+                                      '/http(.*)no2fa-provider.de/'
+                                  ),
+        /**
+         *  Per value in entityids, you may specify another set of regular expressions to match the 
+         *  attributes in the SAML request. If there is a match in any attribute value, this filter will 
+         *  set the state variable to true.
```

Comment:
OK, I this the logic is a bit difficult to follow, since this filter actually **deactivates** 2FA, which is not quite clear by the name.

So if __no_ ``excludeattribute`` is given and matches, it will set to "false" for a given entityid.
But if an ``EXCLUDEattribute`` is fiven and matches, it will be set to "true" and 2FA will be active.

I personally find this a bit confusing the double negation.

Because bottomline: If "checkEntity" is "successful", 2FA will be disabled.

Maybe we simple need to rename the filter?

Reply:
We already have the checkClientIP filter, which sets the state variable to false for the 'excludeClientIPs'.
Maybe we should rename not the filter but all three options in that sense to setfalseClientIPs, setfalseEntityIDs, settrueAttributes.

Would this be clearer?

Reply:
I renamed the variables in https://github.com/privacyidea/simplesamlphp-module-privacyidea/pull/89/commits/222bb56910b55620cdb84e59b3745c97104e6cc5